Content.Server/Destructible/Thresholds/Behaviors/SpawnEntitiesBehavior.cs

```diff
@@ -37,12 +42,32 @@ public void Execute(EntityUid owner, DestructibleSystem system, EntityUid? cause
                 {
                     var spawned = system.EntityManager.SpawnEntity(entityId, position.Offset(getRandomVector()));
                     system.StackSystem.SetCount(spawned, count);
+
+                    if (!TransferForensics || !system.EntityManager.TryGetComponent<ForensicsComponent>(owner, out var forensicsComponent))
+                        continue;
+                    var comp = system.EntityManager.EnsureComponent<ForensicsComponent>(spawned);
+                    comp.DNAs = forensicsComponent.DNAs;
+
+                    if (!system.Random.Prob(0.4f))
+                        continue;
+                    comp.Fingerprints = forensicsComponent.Fingerprints;
+                    comp.Fibers = forensicsComponent.Fibers;
                 }
                 else
                 {
                     for (var i = 0; i < count; i++)
                     {
-                        system.EntityManager.SpawnEntity(entityId, position.Offset(getRandomVector()));
+                        var spawned = system.EntityManager.SpawnEntity(entityId, position.Offset(getRandomVector()));
+
+                        if (!TransferForensics || !system.EntityManager.TryGetComponent<ForensicsComponent>(owner, out var forensicsComponent))
```

Comment:
i feel like you could deduplicate this a bit if you lifted the 'give forensics' code into some function